### DISCUSSION TITLE
Get extended types

### DISCUSSION THREAD

**Discussion by aalmada** [ethereum-ts/TypeChain#606] [Q&A]
I have a contract that extends `ERC721`. Some of my methods only need portions of it, for example, `IERC721Metadata`. How can I "cast" the generated type to this interface?

**Comment by krzkaczor** [comment]
Do you have ABI and types for the interface?

**Reply by aalmada** [reply]
TypeChain is generating the type for `IERC721Metadata`. I'm using `ethers-v5` and I think I can somehow get the ABI if needed.

**Reply by krzkaczor** [reply]
I think you're hitting this bug that was recently introduced: https://github.com/dethcrypto/TypeChain/issues/601

 You can downgrade to previous major version of TypeChain and you should be simple assign your contract that extends `IERC721Metadata` to variable of type `IERC721Metadata`.

**Comment by aalmada** [comment]
Hi @krzkaczor! 
Given its name, I suspect #648 is related to this and I'm trying to figure out what changed.
How can I use it? Is it included in version 8.0.0?

**Reply by aalmada** [reply]
It works using the `as` operator. It wasn't working in my case because of an unrelated issue. 😨

**Comment by aalmada** [comment]
@krzkaczor I started writing a series of articles about the use of TypeChain with React Query: https://antao-almada.medium.com/using-react-query-to-query-smart-contracts-part-2-8c64647e653b

This made me return to this subject.

I have a `MyToken` smart contract that derives from `ERC721` and `Pausable`. 

If I have a method that takes a parameters of type `Pausable`, I can pass `MyToken` without any explicit cast. 

If I have a method that takes a parameters of type `IERC721Enumerable` then I have to explicitly cast using `contract as unknown as IERC721Enumerable` or `contract as ERC721 as IERC721Enumerable`.

Is this a known limitation?
